src/Microsoft.NetCore.Analyzers/Core/Runtime/DisposeObjectsBeforeLosingScope.cs

```diff
@@ -197,7 +197,7 @@ bool SkipInterproceduralAnalysis(IMethodSymbol invokedMethod)
 
                     bool CanBeDisposable(ITypeSymbol type)
                         => type.SpecialType == SpecialType.System_Object ||
-                            type.DerivesFrom(disposeAnalysisHelper.IDisposable) ||
+                            type.DerivesFrom(disposeAnalysisHelper!.IDisposable) ||
```

Comment:
💡 Seems like a reasonable place to make the containing local function static to avoid the suppression.

Reply:
Same comment as above. I think compiler's flow analysis needs to be improved here to not generate this false positive.